File: app/src/main/java/com/ssz/entry/ui/home/HomePresenter.java
package com.ssz.entry.ui.home;

import com.ssz.baselibrary.app.di.scope.ActivityScope;

import javax.inject.Inject;

/**
 * @author : zsp
 * time : 2019 12 2019/12/23 15:20
 */
@ActivityScope
public class HomePresenter implements IHomeContract.IPresenter {

    private IHomeContract.IView mView;

    @Inject HomePresenter(IHomeContract.IView view){
        this.mView = view;
    }

    @Override
    public void detach() {
        this.mView = null;
    }
}

File: app/src/main/java/com/ssz/studydemo/module/HomePresenter.kt
package com.ssz.studydemo.module

import com.ssz.studydemo.module.home.IHomeContract

/**
 * @author : zsp
 * time : 2019 09 2019/9/17 15:11
 */
class HomePresenter(val view : IHomeContract.IHomeView) : IHomeContract.IHomePresenter {

    init {
        view.mPresenter = this
    }
}